### PR TITLE
chore(test): Implements tests helpers

### PR DESCRIPTION
**Pull request by srascar**
Add tests helpers

### CODE REVIEW

File: Tests/Admin/AdminFormFieldTestTrait.php [hunk 1]
```diff
@@ -0,0 +1,92 @@
+<?php
+/*
+ * This file is part of the Sonata project.
+ * 
+ * (c) 2018 - Sonata
```
Comment (Bladrak) [inline Tests/Admin/AdminFormFieldTestTrait.php]: Is it sonata or ekino project?

File: Tests/Admin/AdminTestCase.php [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+     * Create mocks of all admin default services
+     * and assign them to a property of the class
+     *
+     * @param AdminInterface $admin
+     */
+    protected function mockDefaultServices(AdminInterface $admin)
+    {
+        // Each element is composed of: original_service_key, propertyName, class
+        $defaultServices = [
+            ['model_manager', 'modelManagerMock', ModelManager::class],
```
Comment (Bladrak) [inline Tests/Admin/AdminTestCase.php]: Might be interesting to have this configurable (if we need to override one of the classes for instance)

File: Tests/CommonTestTrait.php [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+namespace SonataHelpers\Tests;
+
+use Symfony\Component\Translation\TranslatorInterface;
+
+/**
+ * Trait CommonTestTrait
+ *
+ * @author Sylvain Rascar <sylvain.rascar@ekino.com>
+ */
+trait CommonTestTrait
```
Comment (Bladrak) [inline Tests/CommonTestTrait.php]: As this is not directly related to sonata, should we put this in this repository or in another one?

File: Tests/CommonTestTrait.php [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+{
+    /**
+     * @param string $class
+     * @param int    $id
+     *
+     * @return mixed
+     */
+    public function getClassWithId($class, $id)
```
Comment (Bladrak) [inline Tests/CommonTestTrait.php]: Given this returns an instance, maybe call it ``getInstanceWithId``?

File: Tests/CommonTestTrait.php [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+    }
+
+    /**
+     * @param string $key
+     * @param array  $params
+     * @param string $domain
+     *
+     * @return string
+     */
+    public function getTranslationString($key, $params, $domain)
```
Comment (Bladrak) [inline Tests/CommonTestTrait.php]: Could be splitted in two traits or classes, with specific responsibilities (one for translation, one for the class instantiation).

File: Tests/Fragments/FragmentFormFieldTestTrait.php [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+use Sonata\ArticleBundle\FragmentService\FragmentServiceInterface;
+use Sonata\CoreBundle\Form\Type\ImmutableArrayType;
+use Symfony\Component\Form\FormBuilderInterface;
+
+/**
+ * Trait FragmentFormFieldTestTrait
+ *
+ * @author Sylvain Rascar <sylvain.rascar@ekino.com>
+ */
+trait FragmentFormFieldTestTrait
```
Comment (Bladrak) [inline Tests/Fragments/FragmentFormFieldTestTrait.php]: Same as previous class, this is not directly related to sonata, might have its place in a specific repo.

File: Tests/Admin/AdminTestCase.php (hunk 2 above)
Comment (srascar) [reply Tests/Admin/AdminTestCase.php]: I don't want it configurable. This function is supposed to simulate the compiler pass of sonata. That's also why `applyDefaults` is private
If you want to override a class, you have access to the property and you have access to the admin to call it's setter

Comment (Bladrak) [reply Tests/Admin/AdminTestCase.php]: Ok, and how would you handle adding another mocked service to the admin? Might be worth enhancing the documentation with a use case for that, if this doesn't allow it.

File: Tests/Fragments/FragmentFormFieldTestTrait.php (hunk 6 above)
Comment (srascar) [reply Tests/Fragments/FragmentFormFieldTestTrait.php]: It is related to SonataArticle

Comment (Bladrak) [reply Tests/Fragments/FragmentFormFieldTestTrait.php]: Yes my bad, missed the fragment part :)

File: Tests/Admin/AdminTestCase.php (hunk 2 above)
Comment (srascar) [reply Tests/Admin/AdminTestCase.php]: Done

**Comment by Bladrak** [comment]
Good to me, great additions, but may be less prioritary than #9

**Comment by bmaziere** [comment]
@srascar should rebase from master

**Comment by bmaziere** [comment]
next approval can merge @Wharenn @qsomazzi @bdejacobet @nanofelis

**Comment by bmaziere** [comment]
up  @Wharenn @qsomazzi @bdejacobet @nanofelis
